iac-common/src/main/java/org/sonar/iac/common/checks/PrivilegeEscalationCheckUtils.java

```diff
@@ -0,0 +1,109 @@
+    new PrivilegeEscalationVector("iam:UpdateLoginProfile"),
+    new PrivilegeEscalationVector("iam:AttachUserPolicy"),
+    new PrivilegeEscalationVector("iam:AttachGroupPolicy"),
+    new PrivilegeEscalationVector("iam:AttachRolePolicy", "sts:AssumeRole"),
+    new PrivilegeEscalationVector("iam:PutUserPolicy"),
+    new PrivilegeEscalationVector("iam:PutGroupPolicy"),
+    new PrivilegeEscalationVector("iam:PutRolePolicy", "sts:AssumeRole"),
+    new PrivilegeEscalationVector("iam:AddUserToGroup"),
+    new PrivilegeEscalationVector("iam:UpdateAssumeRolePolicy", "sts:AssumeRole"),
+    new PrivilegeEscalationVector("iam:PassRole", "ec2:RunInstances"),
+    new PrivilegeEscalationVector("iam:PassRole", "lambda:CreateFunction", "lambda:InvokeFunction"),
+    new PrivilegeEscalationVector("iam:PassRole", "lambda:CreateFunction", "lambda:AddPermission"),
+    new PrivilegeEscalationVector("iam:PassRole", "lambda:CreateFunction", "lambda:CreateEventSourceMapping"),
+    new PrivilegeEscalationVector("iam:PassRole", "cloudformation:CreateStack"),
+    new PrivilegeEscalationVector("iam:PassRole", "datapipeline:CreatePipeline", "datapipeline:PutPipelineDefinition"),
+    new PrivilegeEscalationVector("iam:PassRole", "glue:CreateDevEndpoint"),
+    new PrivilegeEscalationVector("glue:UpdateDevEndpoint"),
+    new PrivilegeEscalationVector("lambda:UpdateFunctionCode")
+  );
+
+  public static boolean escalationVectorApply(Stream<String> actionPermissions) {
+    Set<Permission> permissionVector = actionPermissions.map(Permission::of).collect(Collectors.toSet());
+    return ESCALATION_VECTORS.stream().anyMatch(vector -> vector.isSubsetOf(permissionVector));
+  }
+
+  public static class PrivilegeEscalationVector {
+
+    private final List<Permission.SimplePermission> permissions;
+
+    public PrivilegeEscalationVector(String... permissions) {
+      this.permissions =  Stream.of(permissions).map(Permission.SimplePermission::new).collect(Collectors.toList());
+    }
+
+    public boolean isSubsetOf(Collection<Permission> actionPermissions) {
+      return permissions.stream().allMatch(p -> actionPermissions.stream().anyMatch(p::isCoveredBy));
+    }
+  }
+
+  public abstract static class Permission {
+    protected final String permissionName;
+
+    protected Permission(String permissionName) {
+      this.permissionName = permissionName;
+    }
+
+    public static Permission of(String permissionName) {
+      return permissionName.endsWith("*") ? new WildCardPermission(permissionName) : new SimplePermission(permissionName);
+    }
+
+    static class SimplePermission extends Permission {
+
+      protected SimplePermission(String permissionName) {
+        super(permissionName);
+      }
+
+      public boolean isCoveredBy(Permission other) {
+        if (other instanceof WildCardPermission) {
+          return (permissionName.substring(0, permissionName.indexOf(':') + 1) + "*").equals(other.permissionName);
```

Comment:
Minor: we could store the "prefix" of each permission in the constructor.
If we do, we should not crash on an unexpected permission name which wouldn't contain a `:`.

Reply:
I've added a test case to show that we will not crash on unexpected permission formats.